File: php/features.php
<?php
declare(strict_types=1);

// RadioSrf1 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RadioSrf1Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RadioSrf1BaseFeature();
            case "test":
                return new RadioSrf1TestFeature();
            default:
                return new RadioSrf1BaseFeature();
        }
    }
}

File: py/features.py
# RadioSrf1 SDK feature factory

from feature.base_feature import RadioSrf1BaseFeature
from feature.test_feature import RadioSrf1TestFeature


def _make_feature(name):
    features = {
        "base": lambda: RadioSrf1BaseFeature(),
        "test": lambda: RadioSrf1TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
